t/t2035-checkout-locked.sh

```diff
@@ -0,0 +1,44 @@
+#!/bin/sh
+
+test_description='checkout must be able to overwrite open files'
+. ./test-lib.sh
+
+test_expect_success 'setup' '
+
+	echo hello >world &&
+	mkdir dear &&
+	echo hello>dear/world &&
+	git add -A &&
+	git commit -m hi &&
+	git tag hi &&
+	git checkout -b bye &&
+	git mv dear cruel &&
+	echo goodbye >world &&
+	echo goodbye >cruel/world &&
+	git add -A &&
+	git commit -m bye
+'
+
+test_expect_success 'lock for read ' '
```

Comment:
The file is not locked, only opened for reading. Maybe `checkout overwrites open file`?

Reply:
Done